EmbodiedSuffering_Engine/Compute/SufferingIndex.cs

```diff
@@ -0,0 +1,115 @@
+
+        [Description("Returns the Suffering Index based on Material Imports data.")]
+        [Input("numberOfEnslavedPeople", "A list of values from the GlobalSlaveryIndex dataset for the country/countries you wish to evaluate.")]
+        [Input("materialImportSource", "The MaterialImportSources dataset or object used to calculate the resultant index value.")]
+        [Input("acceptableThreshold", "The acceptable value of Number of Enslaved People to be considered in the calculation. Values found in the NumberOfEnslavedPeople input above this threshold will not be considered.")]
+        [MultiOutput(0, "sufferingIndex", "The SufferingIndex per import location and slavery data.")]
+        [MultiOutput(1, "culledCountries", "Countries removed from the calculation due to insufficient data.")]
+        [MultiOutput(2, "invalidCountries", "Countries that had no data for NumberOfEnslavedPeople.")]
+        public static Output<double, List<string>, List<string>> SufferingIndex(List<double> numberOfEnslavedPeople, MaterialImportSources materialImportSources, double acceptableThreshold)
+        {
+            double sufferingIndex = 0;
+            List<double> importRatios = materialImportSources.ImportRatios.ToList();
+            List<List<Country>> exportCountries = new List<List<Country>>();
+            List<string> culledCountries = new List<string>();
+            List<string> invalidCountries = new List<string>();
+
+            // List length check: Fail the calculation if the lists are of unequal lengths
+            if (numberOfEnslavedPeople.Count != importRatios.Count)
+            {
+                int numCount = numberOfEnslavedPeople.Count;
+                int ratiosCount = importRatios.Count;
+                BH.Engine.Base.Compute.RecordError("The provided list's lengths to do correspond as expected." + $"\nNumberOfEnslavedPeople count = {numCount}" + $"\nImportRatios count = {ratiosCount}. \nPlease update your data and try again.");
+            } else
+            {
+                // Remove after debugging is complete
+                BH.Engine.Base.Compute.RecordNote("Lists are the same length, proceeding to threshold check.");
+            }
+
+            // Add countries to ImportCountries[]
+            for (int i = 0; i < materialImportSources.ImportRatios.Count; i++)
+            {
+                exportCountries.Add(materialImportSources.ExportCountries);
+            }
+
+            // Pass a list of string import countries
+            // This is used in the threshold check
+            List<string> importCountriesAsStrings = exportCountries.ConvertAll(x => x.ToString());
+
+            // Threshold check: Toss anything above the input threshold into badCountries[]
+            for (int i = 0; i < numberOfEnslavedPeople.Count; i++)
+            {
+                if (numberOfEnslavedPeople[i] > acceptableThreshold)
+                {
+                    culledCountries.Add(importCountriesAsStrings[i]);
```

Comment:
@FraserGreenroyd this is functioning, however I'm seeing a casting to generic list of type County (enum) which is not useful in the proposed workflow. Is there a way here that we can cast to string as intended? 

See line 75 where I convert each enum to a string, then add them to a list of strings on line 82.